internal/step/foreach/provider.go

```diff
@@ -212,18 +195,35 @@ func (l *forEachProvider) LoadSchema(inputs map[string]any, workflowContext map[
 	}
 
 	return &runnableStep{
-		workflow:    preparedWorkflow,
-		parallelism: inputs["parallelism"].(int64),
-		logger:      l.logger,
+		workflow: preparedWorkflow,
+		logger:   l.logger,
 	}, nil
 }
 
 type runnableStep struct {
-	workflow    workflow.ExecutableWorkflow
-	parallelism int64
-	logger      log.Logger
+	workflow workflow.ExecutableWorkflow
+	logger   log.Logger
 }
 
+var parallelismSchema = schema.NewPropertySchema(
+	schema.NewIntSchema(
+		schema.PointerTo[int64](1),
+		nil,
+		nil,
+	),
```

Comment:
Since it doesn't make sense to have a _negative_ parallelism value, would it be better (e.g., improve usability) if the value had an unsigned type?

Reply:
The schema uses signed 64 bit integers. I'll add a minimum value to the schema.

Reply:
Actually, it already has a minimum of 1, so it's correct.

Reply:
Yeah, a problem with the schema constructors is that there are so many undistinguished parameters that it can be really hard to figure out what they're declaring without searching for the type definition, which is never in the PR. 😆

Reply:
Yeah. It's a pain to read when you don't have help from the IDE.